### ISSUE TITLE
Gtk: Crash after toggling Output pane font checkbox

### ISSUE THREAD

**Issue by danpla** [manisandro/gImageReader#111]
Gtk version crashes when I toggle the Default checkbox of the Output pane font option in the Preferences.

It happens almost every time, but not always, so you'll probably need to start the program more than once and toggle the checkbox several times to reproduce the issue.

The console output is:
```
unhandled exception (type Glib::Error) in signal handler:
domain: g-exec-error-quark
code  : 8
what  : Failed to execute child process "gimagereader-gtk" (No such file or directory)
```
I think first lines of the GDB's backtrace will be more useful:
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7256098 in Gtk::Builder::get_cwidget(Glib::ustring const&) ()
   from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
(gdb) backtrace
#0  0x00007ffff7256098 in Gtk::Builder::get_cwidget(Glib::ustring const&) ()
   from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#1  0x00007ffff725616f in Gtk::Builder::get_widget_checked(Glib::ustring const&, unsigned long) () from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#2  0x0000000000470da5 in get_widget<Gtk::Widget> (widget=<synthetic pointer>, 
    name=..., this=<optimized out>)
    at /usr/include/gtkmm-3.0/gtkmm/builder.h:585
#3  operator() (name=..., this=0xe6dbb8)
    at /build/gimagereader-OCdzmA/gimagereader-3.1.99/gtk/src/common.hh:77
#4  OutputEditorHOCR::setFont (this=0xe6da10)
    at /build/gimagereader-OCdzmA/gimagereader-3.1.99/gtk/src/OutputEditorHOCR.cc:427
#5  0x00007ffff5c27038 in Glib::SignalProxyNormal::slot0_void_callback(_GObject*, void*) () from /usr/lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#6  0x00007ffff4d343b8 in g_closure_invoke ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#7  0x00007ffff4d45fd1 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff4d4da29 in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff4d4dce2 in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00007ffff60c0f6f in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#11 0x00007ffff7258340 in Gtk::Button_Class::clicked_callback(_GtkButton*) ()
   from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#12 0x00007ffff4d345e7 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff4d4d088 in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff4d4dce2 in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff60c097d in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#16 0x00007ffff72582a0 in Gtk::Button_Class::released_callback(_GtkButton*) ()
   from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#17 0x00007ffff4d345e7 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#18 0x00007ffff4d4d088 in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#19 0x00007ffff4d4dce2 in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#20 0x00007ffff5f1c6a1 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#21 0x00007ffff72ecf14 in Gtk::Widget::on_button_release_event(_GdkEventButton*) () from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#22 0x00007ffff72ef0b4 in Gtk::Widget_Class::button_release_event_callback(_GtkWidget*, _GdkEventButton*) () from /usr/lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#23 0x00007ffff5fe7d3e in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#24 0x00007ffff4d3455f in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#25 0x00007ffff4d4d088 in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#26 0x00007ffff4d4dce2 in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#27 0x00007ffff6110974 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#28 0x00007ffff5fe608c in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#29 0x00007ffff5fe791a in gtk_main_do_event ()
   from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#30 0x00007ffff575ac22 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#31 0x00007ffff4a64e04 in g_main_context_dispatch ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
...
```

Xubuntu 14.04 and 16.04, x86_64
gImageReader 3.1.99

**Comment by manisandro** [comment]
The output on the console actually seems to indicate that the crash handler is broken, which would be an additional issue.

**Comment by manisandro** [comment]
Crash should be fixed. Did you actually ever see the crash handler dialog?

**Comment by danpla** [comment]
I compiled from source: everything is OK now.

I forget to say that this issue appeared only in Gtk version.

Yes, I saw the crash handler in Qt version (#108), but not in Gtk.

In this particular case, the problem is that `pkgExePath` initialized directly from argv[0]. When I run `/usr/bin/gimagereader-gtk` instead of just `gimagereader-gtk`, the crash handler appears - the exception message above is pretty informative.

It seems like the correct way is to look in `/proc`, like in `get_application_dir`, and fall back to argv[0] on errors.
